src/main/java/org/folio/circulation/resources/CirculationRulesResource.java

```diff
@@ -115,14 +146,78 @@ private void put(RoutingContext routingContext) {
     LoanCirculationRulesEngineResource.clearCache(new WebContext(routingContext).getTenantId());
     RequestCirculationRulesEngineResource.clearCache(new WebContext(routingContext).getTenantId());
 
-    loansRulesClient.put(rulesInput.copy()).thenAccept(result ->
-      result.applySideEffect(response -> {
+    clients.circulationRulesStorage().put(rulesInput.copy())
+      .thenAccept(res -> res.applySideEffect(response -> {
         if (response.getStatusCode() == 204) {
           SuccessResponse.noContent(routingContext.response());
         } else {
           ForwardResponse.forward(routingContext.response(), response);
         }
-    }, cause -> cause.writeTo(routingContext.response())));
+      }, cause -> cause.writeTo(routingContext.response())));
+  }
+
+   private boolean isPolicyExisting(Map<String, Set<String>> existingPolicyIds,
+    String policyType, List<CirculationRulesParser.PolicyContext> policies) {
+
+    return existingPolicyIds != null && isNotEmpty(existingPolicyIds.get(policyType))
```

Comment:
How about using `MapUtils.isNotEmpty(existingPolicyIds)` instead of `existingPolicyIds != null`?